src/core/build.jl

```diff
@@ -92,6 +92,100 @@ function _instantiate_vars(md::ModelDef)
     return vdict
 end
 
+"""
+    _find_paths_and_names(obj::AbstractComponentDef, datum_name::Symbol)
+
+Recurses through sub components and finds the full path(s) to desired datum, and their
+names at the leaf level. Returns a tuple (paths::Vector{ComponentPath}, datum_names::Vector{Symbol})
+"""
+function _find_paths_and_names(obj::AbstractComponentDef, datum_name::Symbol)
+
+    # Base case-- leaf component
+    if obj isa ComponentDef
+        return ([nothing], [datum_name])
+    end
```

Comment:
note to self: might clean up this base case to actually return the comp name instead of nothing, after the comp_path bug is fixed separately